pkg/backingstore/reconciler.go

```diff
@@ -303,7 +303,19 @@ func (r *Reconciler) ReconcileDeletion() error {
 		}
 		err := r.NBClient.DeletePoolAPI(nb.DeletePoolParams{Name: r.PoolInfo.Name})
 		if err != nil {
-			return err
+			if rpcErr, isRPCErr := err.(*nb.RPCError); isRPCErr {
+				if rpcErr.RPCCode == "DEFAULT_RESOURCE" {
+					return util.NewPersistentError("DefaultResource",
+						fmt.Sprintf("DeletePoolAPI cannot complete because pool %q is an account default resource", r.PoolInfo.Name))
+				} else if rpcErr.RPCCode == "IN_USE" {
+					return util.NewPersistentError("ResourceInUse",
+						fmt.Sprintf("DeletePoolAPI cannot complete because pool %q has buckets attached", r.PoolInfo.Name))
+				} else {
```

Comment:
How about just removing these two `else` statements and just keep it as a fallback to `return err` in any other case?

Reply:
changed to switch

Reply:
removed the else as recommended